### PR TITLE
Pil 723: RFM security question validation

### PR DESCRIPTION
**Pull request by alikouhestani**

### CODE REVIEW

File: app/views/rfm/SecurityQuestionsNoMatchView.scala.html [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+)
+
+@()(implicit request: Request[_], appConfig: FrontendAppConfig, messages: Messages)
+
+@layout(
+    pageTitle    = titleNoForm(messages("rfm.security.errorPage.title")),
+    showBackLink = false
+) {
+
+        <h1 class="govuk-heading-l">@messages("rfm.security.errorPage.heading")</h1>
```
Comment (jassalrichy) [inline app/views/rfm/SecurityQuestionsNoMatchView.scala.html:30]: use gds components: heading, paragraph

File: app/views/rfm/SecurityQuestionsCheckYourAnswersView.scala.html [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+)
+
+@(list: SummaryList)(implicit request: Request[_], appConfig: FrontendAppConfig, messages: Messages)
+
+@layout(pageTitle = titleNoForm(messages("securityQuestionsCheckYourAnswers.title"))) {
+
+    @formHelper(action = controllers.rfm.routes.SecurityQuestionsCheckYourAnswersController.onSubmit) {
+
+    <span class="govuk-caption-l">@messages("securityQuestionsCheckYourAnswers.heading.caption")</span>
+    <h1 class="govuk-heading-l">@messages("securityQuestionsCheckYourAnswers.heading")</h1>
```
Comment (jassalrichy) [inline app/views/rfm/SecurityQuestionsCheckYourAnswersView.scala.html:36]: use gds components: sectionHeader, heading

File: app/views/rfm/SecurityQuestionsNoMatchView.scala.html [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+@this(
+    layout: templates.Layout,
+    govukButton: GovukButton
+)
+
+@()(implicit request: Request[_], appConfig: FrontendAppConfig, messages: Messages)
+
+@layout(
+    pageTitle    = titleNoForm(messages("rfm.security.errorPage.title")),
+    showBackLink = false
```
Comment (jassalrichy) [inline app/views/rfm/SecurityQuestionsNoMatchView.scala.html:27]: missing parameter 'authorised = false' which will redirect the user to gov.uk page when clicking the service banner link.

File: app/views/rfm/SecurityQuestionsNoMatchView.scala.html [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+) {
+
+        <h1 class="govuk-heading-l">@messages("rfm.security.errorPage.heading")</h1>
+        <p class="govuk-body govuk-!-margin-bottom-5">@messages("rfm.security.errorPage.p1")</p>
+                    <p class="govuk-body">
+                        <a> @messages("rfm.security.errorPage.p2")</a>
+             <a href="@controllers.routes.UnderConstructionController.onPageLoad" class="govuk-link govuk-link--no-visited-state"> @messages("rfm.security.errorPage.p2.link")</a>
```
Comment (jassalrichy) [inline app/views/rfm/SecurityQuestionsNoMatchView.scala.html:34]: 'go back to try again using different details' link should redirect the User to the first security question.

File: app/controllers/rfm/SecurityQuestionsNoMatchController.scala [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import scala.concurrent.ExecutionContext
+
+class SecurityQuestionsNoMatchController @Inject() (val controllerComponents: MessagesControllerComponents, view: SecurityQuestionsNoMatchView)(
+  implicit appConfig:                                                         FrontendAppConfig
+) extends FrontendBaseController
+    with I18nSupport {
+
+  def onPageLoad: Action[AnyContent] = Action { implicit request =>
+    Ok(view())
+  }
```
Comment (jassalrichy) [inline app/controllers/rfm/SecurityQuestionsNoMatchController.scala:35]: Need a spec for this controller.

File: app/controllers/rfm/SecurityQuestionsCheckYourAnswersController.scala [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+package controllers.rfm
+
+import cats.data.OptionT
+import cats.implicits.catsSyntaxApplicativeError
+import com.google.inject.Inject
+import config.FrontendAppConfig
+import connectors.UserAnswersConnectors
+import controllers.actions.{DataRequiredAction, DataRetrievalAction, RfmIdentifierAction}
+import controllers.routes
+import models.{InternalIssueError, Mode}
```
Comment (jassalrichy) [inline app/controllers/rfm/SecurityQuestionsCheckYourAnswersController.scala:26]: remove unused imports: models.Mode, pages.plrReferencePage

File: app/views/rfm/SecurityQuestionsNoMatchView.scala.html (hunk 1 above)
Comment (alikouhestani) [reply app/views/rfm/SecurityQuestionsNoMatchView.scala.html:30]: Done

File: app/views/rfm/SecurityQuestionsNoMatchView.scala.html (hunk 3 above)
Comment (alikouhestani) [reply app/views/rfm/SecurityQuestionsNoMatchView.scala.html:27]: Done

File: app/views/rfm/SecurityQuestionsNoMatchView.scala.html (hunk 4 above)
Comment (alikouhestani) [reply app/views/rfm/SecurityQuestionsNoMatchView.scala.html:34]: Done

File: app/controllers/rfm/SecurityQuestionsCheckYourAnswersController.scala (hunk 6 above)
Comment (alikouhestani) [reply app/controllers/rfm/SecurityQuestionsCheckYourAnswersController.scala:26]: Done

File: app/views/components/gds/paragraphWithLink.scala.html [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+ * limitations under the License.
+ *@
+
+@this()
+
+@(text: String,  call: Call, hyperLinkMessage: String, classes: String = "govuk-link govuk-link--no-visited-state", id: Option[String] = None)
+
+<p class="govuk-body">
+<a> @text</a> <a href="@call" class="@classes" @id.map{id => id="@id"}> @hyperLinkMessage</a>
+</p>
```
Comment (jassalrichy) [inline app/views/components/gds/paragraphWithLink.scala.html:23]: add a new line at the end

File: app/views/rfm/SecurityQuestionsCheckYourAnswersView.scala.html [hunk 8]
```diff
@@ -18,31 +18,29 @@
...
     @formHelper(action = controllers.rfm.routes.SecurityQuestionsCheckYourAnswersController.onSubmit) {
-
-    <span class="govuk-caption-l">@messages("securityQuestionsCheckYourAnswers.heading.caption")</span>
-    <h1 class="govuk-heading-l">@messages("securityQuestionsCheckYourAnswers.heading")</h1>
-
+    @sectionHeader(messages("securityQuestionsCheckYourAnswers.heading.caption"))
+    @heading(messages("securityQuestionsCheckYourAnswers.heading"), "govuk-heading-l")
     @govukSummaryList(list)
-
     <br>
```
Comment (jassalrichy) [inline app/views/rfm/SecurityQuestionsCheckYourAnswersView.scala.html:40]: remove <br> and add .withCssClass("govuk1-!-margin-top-6") to the ButtonViewModel below

File: app/views/components/gds/paragraphWithLink.scala.html (hunk 7 above)
Comment (jassalrichy) [inline app/views/components/gds/paragraphWithLink.scala.html:23]: add new line at end of file

**Review by jassalrichy — DISMISSED** [review]

**Review by jassalrichy — DISMISSED** [review]

**Pull request closed without merging**